socketcan_interface/include/socketcan_interface/socketcan.h

```diff
@@ -200,7 +203,7 @@ class SocketCANInterface : public AsioDriver<boost::asio::posix::stream_descript
                 input_.id = frame_.can_id & CAN_EFF_MASK;
                 input_.is_error = 1;
 
-                LOG("error: " << input_.id);
+                ROSCANOPEN_ERROR("socketcan_interface", "error frame: " << input_.id);
```

Comment:
Why do you limit the output to the id? The subsequent message in can_layer prints the whole frame:
```
[ERROR] [1562238561.394633541]: error frame: 4
[ERROR] [1562238561.394850816]: Received error frame: 20000000#0030000000000000
```

Reply:
I just batch-replaced, but it sounds resonable to print the full frame :)

Reply:
What confuses my right now: shouldn't the second line be `20000004`?

Reply:
IMHO the first message could get dropped completely

Reply:
actually I disconnected the CAN-bus, so the error is `could not reset node '3'`.
Candump shows nothing besides `(1562591423.558640)  can0  TX - -  000   [2]  82 03` and thus I am unsure, what the error telegram with ID 4 says. Is there an internal error message generated by socketcan?

Reply:
Yes I agree, that the first error message could be dropped. It is just duplicated information (once the second output is interpreted correctly).

Reply:
The output with 5889ab66fa21a11f98cfc7f08451d7df4906b2d7 and this simple test case now reads:
```
[ERROR] [1564589661.190719770]: internal error: 4
[ INFO] [1564589661.190777365]: Current state: 2 device error: system:0 internal_error: 4 (controller problems;)
[ INFO] [1564589661.190813729]: Current state: 1 device error: system:0 internal_error: 4 (controller problems;)
[ERROR] [1564589661.191039114]: Received error frame: 20000004#0030000000000000
[ERROR] [1564589661.198288816]: CAN not ready
```